spec/eof.md

```diff
@@ -72,8 +102,9 @@ The following validity constraints are placed on the container format:
 - `code_size` may not be 0
 - the number of container sections must not exceed 256
 - `container_size` may not be 0, but container sections are optional
-- the total size of the container without container sections must be `13 + 2*num_code_sections + types_size + code_size[0] + ... + code_size[num_code_sections-1] + data_size`
-- the total size of the container with at least one container section must be `16 + 2*num_code_sections + types_size + code_size[0] + ... + code_size[num_code_sections-1] + data_size + 2*num_container_sections + container_size[0] + ... + container_size[num_container_sections-1]`
+- the total size of a deployed container without container sections must be `13 + 2*num_code_sections + types_size + code_size[0] + ... + code_size[num_code_sections-1] + data_size`
+- the total size of a deployed container with at least one container section must be `16 + 2*num_code_sections + types_size + code_size[0] + ... + code_size[num_code_sections-1] + data_size + 2*num_container_sections + container_size[0] + ... + container_size[num_container_sections-1]`
+- the total size of not yet deployed container might be less than the above values due to how the data section is rewritten and resized during deployment (see [Data Section Lifecycle](#data-section-lifecycle))
```

Comment:
It might be less, but it may not be arbitrarily less. Quite important validation check is also that total size may not be less than `sum_of_all_sections - data_size`, i.e. only data section may be truncated, but nothing else.

Reply:
```suggestion
- the total size of not yet deployed container might be up to `data_size` less than the above values due to how the data section is rewritten and resized during deployment (see [Data Section Lifecycle](#data-section-lifecycle))
```